docs/BUILD.md

```diff
@@ -1,6 +1,56 @@
 # Building s2n-tls
 
-s2n-tls can be built as follows:
+To use s2n-tls, you must build the library from the source and then include it in your program.
+## Requirements
+
+s2n-tls supports and tests on **x86** and **arm** architectures.
+
+### System requirements
+
+* 20GB RAM available
+
+### Supported OS and Distributions: 
+
+| OS     | Distros |      |        |
+|--------|---------|------|--------|
+| Redhat | Fedora  | AL   |
+| debian | ubuntu  |
```

Comment:
Do we test on debian?

Reply:
The [os-release](https://man7.org/linux/man-pages/man5/os-release.5.html) man page talks about 'derivatives of', with a specific field tracking these, `ID_LIKE`.  Ubuntu is derived from Debian, which is itself an OS, which we do not specifically test on.

example from AL2:
```grep ID /etc/os-release
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
```

Ubuntu22:
```
grep ID /etc/os-release                                                                                                                                                                    [21:09:06 2023-09-25]
VERSION_ID="22.04"
ID=ubuntu
ID_LIKE=debian
```